sys/riscv/include/pte.h

```diff
@@ -66,6 +66,17 @@ typedef	uint64_t	pn_t;			/* page number */
 #define	Ln_ENTRIES	(1 << Ln_ENTRIES_SHIFT)
 #define	Ln_ADDR_MASK	(Ln_ENTRIES - 1)
 
+#ifdef CPU_CHERI
+/* CHERI uses reserved bits in 55:63 */
+#define	PTE_SC		(1UL << 63) /* Store Capability */
+#define	PTE_LC		(1UL << 62) /* Load Capability */
+#define	PTE_PROMOTE_CHERI (PTE_LC | PTE_SC)
+#define	PTE_KERN_CAP	(PTE_KERN | PTE_LC | PTE_SC)
```

Comment:
Weird to define this before PTE_KERN. If you instead call PTE_PROMOTE_CHERI PTE_LCSC or something then you can still use it in PTE_PROMOTE, but also have a single definition for PTE_KERN_CAP below PTE_KERN.

Reply:
It is a bit weird, but also the beauty of cpp. :-P  (Not really, it's mostly weird)

One thing to keep in mind is that revocation will probably need to add the other 3 bits to `PTE_PROMOTE_CHERI` but not to `PTE_KERN_CAP`.  What I could do perhaps is what I almost did originally which is to add another #ifdef and define `PTE_KERN_CAP` below `PTE_KERN`.

Reply:
Or maybe a PTE_KERN_CHERI that is conditionally defined that ends up in PTE_KERN_CAP?